Xwt/Xwt.Drawing/Image.cs

```diff
@@ -967,36 +973,123 @@ public void ReleaseReference (bool disposing)
 		public NativeImageRef NextRef { get; set; }
 	}
 
-	class ImageTagSet
+	sealed class ImageTagCache
+	{
+		/* Some stats from an app using Xwt:
+1474 dark
+1304 contrast
+1296 contrast~dark
+ 846 sel
+ 846 dark~sel
+ 132 disabled
+ 120 dark~disabled
+ 116 contrast~disabled
+ 116 contrast~dark~disabled
+  22 error
+  22 dark~error
+  22 contrast~error
+  22 contrast~dark~error
+  14 contrast~dark~sel
+  12 contrast~sel
+   6 disabled~dark
+   6 dark~contrast
+   2 sel~error
+   2 pressed~dark
+   2 pressed
+   2 hover~dark
+   2 hover
+   2 dark~sel~error
+   2 contrast~sel~error
+   2 active~sel
+   2 active~dark~sel
+   2 active~dark
+   2 active~contrast~dark
+   2 active~contrast
+   2 active
+
+		Keep in sync with knownTagArrays.
+		These tag items amount for 97% of the image tags found in images.
+		*/
+		readonly string[] knownTags = new[] {
+			"dark",
```

Comment:
I don't know why I made the cache this way. Looking back, I think that we can store the original `tags` string and cache the ImageTagSet.

Reply:
Basically, we can have some presets, but cache tags that are added later as well.